compiler/luci/pass/src/FoldShapePass.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FoldShapePass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+namespace
+{
+
+template <loco::DataType OutType> luci::CircleConst *folding(luci::CircleShape *shape)
+{
+  auto input_node = loco::must_cast<luci::CircleNode *>(shape->input());
+  auto name = input_node->name();
+  auto shape_status = input_node->shape_status();
+  auto rank = input_node->rank();
+  assert(name.length() > 0);
+  if (shape_status != luci::ShapeStatus::VALID)
+    return nullptr;
+  if (rank <= 0)
+    return nullptr;
+  for (uint32_t i = 0; i < rank; i++)
+  {
+    auto dim = input_node->dim(i);
+    if (!dim.known()) // If dim is unknown, return nullptr
```

Comment:
I don't think `// If dim is unknown, return nullptr` is necessary.
It just explains the code itself.